eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -146,7 +146,7 @@ object Commitments {
     // the HTLC we are about to create is outgoing, but from their point of view it is incoming
     val outgoingHtlcs = reduced.htlcs.filter(_.direction == IN)
 
-    val htlcValueInFlight = UInt64(outgoingHtlcs.map(_.add.amountMsat).sum)
+    val htlcValueInFlight = UInt64(outgoingHtlcs.map(_.add.amountMsat.toLong).sum)
```

Comment:
We should be able to write `outgoingHtlcs.map(_.add.amountMsat).sum` here (all that is needed is an implementation of `Numeric[MilliSatoshi]`, it's missing in bitcoin-lib but can be added here for now)

Reply:
How about adding `Numeric[MilliSatoshi]` to https://github.com/ACINQ/bitcoin-lib/pull/36 ? Then i can build this PR on top of the upgraded bitcoin-lib.